src/main/java/frc/robot/systems/ElevatorWristFSM.java

```diff
@@ -0,0 +1,212 @@
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.SparkMaxLimitSwitch;
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+
+public class ElevatorWristFSM {
+	/* ======================== Constants ======================== */
+	// FSM state definitions
+	private enum FSMState {
+		MOVING_IN,
+		MOVING_OUT,
+		ZEROING,
+		IDLE
+	}
+	private static final double ZEROING_SPEED = -0.1;
+	private static final double PID_CONSTANT_WRIST_P = 0.001;
+	private static final double PID_CONSTANT_WRIST_I = 0.00000001;
+	private static final double PID_CONSTANT_WRIST_D = 0.00000001;
+	private static final double OUTER_LIMIT_ENCODER = 100.0; //subject to change based on testing
+	private static final float MAX_UP_POWER = 0.2f;
+	private static final float MAX_DOWN_POWER = -0.2f;
+	private static final double WRIST_IN_ENCODER_ROTATIONS = -1;
+	private static final double WRIST_OUT_ENCODER_ROTATIONS = 50;
+
+	/* ======================== Private variables ======================== */
+	private FSMState currentState;
+	// Hardware devices should be owned by one and only one system. They must
+	// be private to their owner system and may not be used elsewhere.
+	private CANSparkMax wristMotor;
+	private SparkMaxPIDController pidControllerWrist;
+	private SparkMaxLimitSwitch wristLimitSwitch;
+	private double currentEncoder;
+	/* ======================== Constructor ======================== */
+	/**
+	 * Create FSMSystem and initialize to starting state. Also perform any
+	 * one-time initialization or configuration of hardware required. Note
+	 * the constructor is called only once when the robot boots.
+	 */
+	public ElevatorWristFSM() {
+		wristMotor = new CANSparkMax(HardwareMap.CAN_ID_WRIST_MOTOR,
+				CANSparkMax.MotorType.kBrushless);
+		wristLimitSwitch = wristMotor.getReverseLimitSwitch(
+				SparkMaxLimitSwitch.Type.kNormallyClosed);
+		wristLimitSwitch.enableLimitSwitch(true);
+		pidControllerWrist = wristMotor.getPIDController();
+		wristMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
+		pidControllerWrist.setP(PID_CONSTANT_WRIST_P);
+		pidControllerWrist.setI(PID_CONSTANT_WRIST_I);
+		pidControllerWrist.setD(PID_CONSTANT_WRIST_D);
+		pidControllerWrist.setOutputRange(MAX_DOWN_POWER, MAX_UP_POWER);
+		// Reset state machine
+		reset();
+	}
+
+	/* ======================== Public methods ======================== */
+	/**
+	 * Return current FSM state.
+	 * @return Current FSM state
+	 */
+	public FSMState getCurrentState() {
+		return currentState;
+	}
+	/**
+	 * Reset this system to its start state. This may be called from mode init
+	 * when the robot is enabled.
+	 *
+	 * Note this is distinct from the one-time initialization in the constructor
+	 * as it may be called multiple times in a boot cycle,
+	 * Ex. if the robot is enabled, disabled, then reenabled.
+	 */
+	public void reset() {
+		currentState = FSMState.IDLE;
+		wristMotor.getEncoder().setPosition(0);
+		currentEncoder = 0;
```

Comment:
I don't think it's necessary to cache the encoder value in a dedicated variable, just call wristMotor.getEncoder().getPosition() directly. That way we always get the most up to date value and we avoid bugs from forgetting to update the private variable.

Reply:
It is cached because we are not updating that current encoder value in IDLE, so we kept a seperate variable to track the changes for all states but IDLE to avoid error.